### PR TITLE
feat: Python Helper POC

### PR DESCRIPTION
**Pull request by JenniferMah**
Initial proof of concept for twilio-python helper library.
Jira: https://issues.corp.twilio.com/browse/DI-2358

### CODE REVIEW

File: src/test/java/com/twilio/oai/TwilioGeneratorTest.java [hunk 1]
```diff
@@ -1,6 +1,7 @@
 package com.twilio.oai;
 
 import java.io.File;
+import java.util.ArrayList;
```
Comment (childish-sambino) [inline src/test/java/com/twilio/oai/TwilioGeneratorTest.java:4]: This needed?

File: src/main/resources/META-INF/services/org.openapitools.codegen.CodegenConfig [hunk 2]
```diff
@@ -3,4 +3,5 @@ com.twilio.oai.TwilioGoGenerator
 com.twilio.oai.TwilioJavaGenerator
 com.twilio.oai.TwilioNodeGenerator
 com.twilio.oai.TwilioPhpGenerator
+com.twilio.oai.TwilioPythonGenerator
```
Comment (childish-sambino) [inline src/main/resources/META-INF/services/org.openapitools.codegen.CodegenConfig:6]: ty for maintaining sorting order!

File: src/main/resources/twilio-python/api-single.mustache [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+        :rtype: str
+        """
+        return '<Twilio.Api.{{apiVersionClass}}.{{resourceName}}>'
+
+{{#responseModel}}
+class {{name}}Instance(InstanceResource):
+    def __init__(self, version, payload{{#resourcePathParams}}, {{paramName}}: {{{dataType}}}{{/resourcePathParams}}):
+        super({{name}}Instance, self).__init__(version)
+        self._properties = { {{#vars}}
+            '{{name}}' = payload.get('{{baseName}}'),{{/vars}}
```
Comment (childish-sambino) [inline src/main/resources/twilio-python/api-single.mustache:83]: Probably need triple brackets for `baseName`.

File: src/main/java/com/twilio/oai/TwilioPythonGenerator.java [hunk 4]
```diff
@@ -0,0 +1,334 @@
...
+    private void addModel(final Map<String, CodegenModel> models, final String dataType) {
+        getModel(dataType).ifPresent(model -> {
+            if (models.putIfAbsent(model.getClassname(), model) == null) {
+                model.getVars().forEach(property -> addModel(models, property.dataType));
+            }
+        });
+    }
+    private Optional<CodegenModel> getModel(final String modelName) {
+        return allModels.stream().filter(model -> model.getClassname().equals(modelName)).findFirst();
+    }
```
Comment (childish-sambino) [inline src/main/java/com/twilio/oai/TwilioPythonGenerator.java:333]: This was moved out of Node into the directory service.

File: src/main/java/com/twilio/oai/TwilioPythonGenerator.java [hunk 5]
```diff
@@ -0,0 +1,334 @@
...
+                    .map(DirectoryStructureService.DependentResource.class::cast)
+                    .forEach(dependent -> {
+                        if (dependent.getType().equals(instanceName)) {
+                            dependent.setType(instanceName + "Import");
+                            dependent.setClassName(instanceName + "Import");
+                            dependent.setImportName(instanceName + " as " + dependent.getType());
+                        }
+                    });
+
+            if (isInstanceOperation || (!hasInstanceOperations )) {
```
Comment (childish-sambino) [inline src/main/java/com/twilio/oai/TwilioPythonGenerator.java:197]: I've a PR up which changes this logic in Node.

**Review by childish-sambino — APPROVED** [review]

**Comment by JenniferMah** [comment]
Merging as a checkpoint for POC python work as discussed on [slack](https://twilio.slack.com/archives/CGPL6QAA0/p1670006332848569).

**Pull request merged**